### PR TITLE
chore: add typecheck for test files

### PR DESCRIPTION
**Pull request by mrazauskas**
Currently only `src` files are type checked. Let’s add type check for test files too?

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "module": "commonjs",
     "moduleResolution": "node",
-    "target": "es2021",
+    "target": "es2019",
```
Comment (SimenB) [inline tsconfig.json:5]: Do we need this? We target node 14, no?

File: src/__tests__/runTest.test.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import runTest from '../run';
 
 jest.mock('tsd-lite');
 jest.mock('../formatter', () => ({
-  ...jest.requireActual('../formatter'),
+  ...(jest.requireActual('../formatter') as typeof import('../formatter')),
```
Comment (SimenB) [inline src/__tests__/runTest.test.ts:17]: No generic?

Comment (mrazauskas) [reply src/__tests__/runTest.test.ts:17]: Right. Only that Jest is not yet published with generic support. Please don’t hurry, we have to revert `toBeCalled` types. Unfortunately they are buggy. I will do this later this week.

File: src/__tests__/runTest.test.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ import runTest from '../run';
 
 jest.mock('tsd-lite');
 jest.mock('../formatter', () => ({
-  ...jest.requireActual('../formatter'),
+  ...jest.requireActual<typeof import('../formatter')>('../formatter'),
```
Comment (mrazauskas) [inline src/__tests__/runTest.test.ts:17]: That’s nicer indeed.

**Comment by SimenB** [comment]
> we have to revert `toBeCalled` types. Unfortunately they are buggy. I will do this later this week.

Oh, missed this 😅 It's been released

File: tsconfig.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "module": "commonjs",
     "moduleResolution": "node",
-    "target": "es2019",
+    "target": "es2020",
```
[the remark this answers is not in the document]
Comment (SimenB) [reply tsconfig.json]: haha, didn't even see [`d9a7edc` (#116)](https://github.com/jest-community/jest-runner-tsd/pull/116/commits/d9a7edca1a3a4b87b60ddd886abade3bcee0ed80) 🙈

File: yarn.lock [hunk 5]
```diff
@@ -4056,22 +4065,33 @@ __metadata:
   languageName: node
   linkType: hard
 
-"jest@npm:^29.0.0":
-  version: 29.0.3
-  resolution: "jest@npm:29.0.3"
+"jest-worker@npm:^29.1.0":
```
Comment (SimenB) [inline yarn.lock:4068]: run `yarn dedupe`

**Review by SimenB — APPROVED** [review]
👍

**Pull request merged**